### DISCUSSION TITLE
API's Base structure Idea 2.0

### DISCUSSION THREAD

**Discussion by Core-LOVE** [Wohlstand/TheXTech#439] [Ideas]
This is revision of my previous structure idea (actually it's pretty similar to first version):

1. **Lua default modules:**
- math (+ math.vector)
- table
- files (basically more user-friendly io)
- string
- coroutine (+ coroutine.thread & coroutine.channel stuff)
- debug
- os
- package (can also be used to register events and etc)
- net (limited socket support?)
- console (i'd recommend to add custom console to thextech as well as module for it. not only it would be good for linux users, but it also sounds awesome he)

(it would be nice if it would be possible to use/require jit or ffi, but it should be somehow limited, idk)

2. **Game's base modules:**
- Assets (Basically a replacement for Graphics.sprites and other stuff)
- Graphics (+ rich/basic text stuff (similar to textplus and etc))
- Audio (Mix of Audio and SFX modules from SMBX2)
- Level
- Episode
- Misc

(Not sure where compatibility and gameinfo stuff could be used. Both seem to fit into Misc and Episode modules)

**Comment by LoveBodhi** [comment]
I suggest "HUD" module, which is for Status Bar Values.

**Comment by LoveBodhi** [comment]
For more extension ideas, see #834 .

**Comment by Shadowblitz16** [comment]
You should not expose libraries that can break out of the sandbox.

All thextech scripts should be safe.
